### ISSUE TITLE
safe-0.3.12 does not compile on GHC 7.0.1

### ISSUE THREAD

**Issue by peti** [ndmitchell/safe#20]
Citing from https://travis-ci.org/peti/hledger-interest/jobs/199950235:

~~~
Safe/Partial.hs:13:19:
    Module ‘GHC.Stack’ does not export ‘CallStack’
~~~

Previous versions of the library did not seem to have this problem.

**Comment by minad** [comment]
Support for GHCs older than 7.4 was dropped since ConstraintKinds are needed, see https://github.com/ndmitchell/safe/issues/18.

@ndmitchell What about re-releasing the current version as 0.4 and marking 0.3.12 as invalid/deprecated?

**Comment by ndmitchell** [comment]
GHC 7.0 is ancient. Is there any reason HLedger still supports it? If there is a legitimate reason I'm happy to release a new version of safe which supports it.

**Comment by peti** [comment]
HLedger supports it mostly because it could. It wasn't a problem. Only it has become one apparently. I don't believe that supporting 7.0.x is important enough to warrant a new release. I do believe, however, that safe's cabal file should declare the required feature so that the build fails at configure time rather than at build time.

**Comment by minad** [comment]
Yes, it is probably better to just add a lower bound on base.

**Comment by ndmitchell** [comment]
I've added a lower bound on base and released safe-0.3.13. Let me know if there are still any problems.

**Comment by minad** [comment]
Maybe you can also edit the bound of 0.3.12 on hackage directly (A new release wouldn't have been necessary)

**Comment by ndmitchell** [comment]
I consider having cryptographically signed git tags which differ from the released contents on Hackage to be somewhat distateful... That said, I've edited 0.3.12 since I guess if I don't then the constraint solver might pick it.

**Comment by minad** [comment]
I agree. Signing is the way to go. I even sign all my git commits these days.

**Comment by nomeata** [comment]
It seems the bound is not low enough:
{{{
Preprocessing library safe-0.3.13...

[1 of 5] Compiling Safe.Partial     ( Safe/Partial.hs, dist/build/Safe/Partial.o )

Safe/Partial.hs:13:19:

    Module ‘GHC.Stack’ does not export ‘CallStack’
}}}
(with 7.10.1)

**Comment by ndmitchell** [comment]
I test on Travis with GHC 7.10.3 - not sure why 7.10.1 is broken...

**Comment by nomeata** [comment]
`CallStack` was added in base 4.8.1.0, which came with 7.10.2.

**Comment by nomeata** [comment]
I guess 2cdbb3fc9a2a2b3600f73cb2f9726b215c33b889 is the culprit, and `GLASGOW_HASKELL` isn’t precise enough.

**Comment by ndmitchell** [comment]
Thanks for the info. I'm away from a computer until the middle of next week but will sort it out then.

**Comment by ndmitchell** [comment]
I've released 0.3.14 which I believe fixes this issue. Please let me know if there are still issues.
